build.py

```diff
@@ -23,18 +27,263 @@ def wait_until_container_is_ready():
     if(p2.match(matched_line)):
         return True
     return False
-    
-def build_and_deploy():
-    subprocess.run(['docker','build','-f','Dockerfile.x86','-t','docker.io/robertbartlettbaron/acct-mgt.x86','.'])
-    subprocess.run(['docker','push','docker.io/robertbartlettbaron/acct-mgt.x86'])
-    subprocess.run(['oc','-n','acct-mgt','rollout','latest','acct-mgt'])
-    result=subprocess.run(['oc','get','all'],stdout=subprocess.PIPE,stderr=subprocess.STDOUT) 
-    return wait_until_container_is_ready()
+
+def build_and_deploy(docker_image):
+    subprocess.run(['docker','build','-f','Dockerfile.x86','-t',docker_image,'.'])
+    subprocess.run(['docker','push',docker_image])
+    subprocess.run(['oc','-n','acct-mgt-2','rollout','latest','acct-mgt-2'])
+    result=subprocess.run(['oc','get','all'],stdout=subprocess.PIPE,stderr=subprocess.STDOUT)
+    return True
+
+def compare_results(result, pattern):
+    if(result is not None):
+        p1=re.compile(pattern)
+        lines=result.stdout.decode('utf-8').split('\n')
+        cnt = 0
+        for l in lines:
+            if(p1.match(l)):
+                return True
+    return False
+
+def oc_service_account_exists(project, service_account):
+    # TODO check privledge
+    result=subprocess.run(['oc', '-n', project, 'get', 'sa', service_account],stdout=subprocess.PIPE,stderr=subprocess.STDOUT)
+    if(compare_results(result,r'^'+service_account)):
+        return True
+    return False
+
+def oc_create_service_account(project, service_account, privledge):
+    subprocess.run(['oc', r'-n', project, 'create', 'sa',service_account])
+    subprocess.run(['oc', 'adm', 'policy', r'add-cluster-role-to-user', privledge, r'system:serviceaccount:' + project + r':' + service_account])
+    return True
+
+
+def oc_project_exists(project):
+    result=subprocess.run(['oc', 'get', 'project', project],stdout=subprocess.PIPE,stderr=subprocess.STDOUT)
+    if(compare_results(result,r'^'+project)):
+        return True
+    return False
```

Comment:
we could just return `compare_results(result,r'^'+project)` instead of having an if statement here

Reply:
this pattern is repeated in other places too.

Reply:
This was written by rob. So, I followed same in my two methods. In case we plan to return something else instead of just true.  "This is the build script creating everything for once."